numpy/core/_add_newdocs.py

```diff
@@ -2035,7 +2035,7 @@
     >>> x.ctypes.data_as(ctypes.POINTER(ctypes.c_long)).contents
     c_long(0)
     >>> x.ctypes.data_as(ctypes.POINTER(ctypes.c_longlong)).contents
-    c_longlong(4294967296L)
+    c_longlong(4294967296)  <- what?
     >>> x.ctypes.shape
```

Comment:
I'm suspicious of this whole doc snippet.

I get `c_longlong(0)` here when I run this locally but I suspect that the doc snippet has changed x to 2**32 or something before running this line.

Reply:
Also I would think this should fail refguide but it doesn't. I'm curious if anyone knows why.

Reply:
This test depends on:

* endianness
* the difference in between `long` and `long long`

The behavior above is what I'd expect from a system with little-endian `long long` and `sizeof(long) *2 == sizeof(long long)` - eg, windows.

Reply:
The right thing to do is remove this part, and perhaps add a comment saying that `ctypes.c_long` is the default numpy integer type.

Reply:
I'm going to drop the L (because it doesn't appear on my system and I believe it's a python2 artifact), and break out the c_uint32 change in another PR.

Reply:
Extracted as #15428